app/User.php

```diff
@@ -182,4 +183,33 @@ public function getGoogle2faSecretAttribute($value)
 
         return decrypt($value);
     }
+
+    /**
+     * Indicate whether the user should be reminded about a reminder or notification.
+     * The user should be reminded only if the date of the reminder matches the
+     * current date, and the current hour matches the hour the account owner
+     * wants to be reminded.
+     *
+     * @param Carbon $date
+     * @return boolean
+     */
+    public function shouldBeReminded(Carbon $date)
+    {
+        $dateOfReminder = $date->hour(0)->minute(0)->second(0)->toDateString();
+
+        $currentDateOnUserTimezone = Carbon::now($this->timezone)->hour(0)->minute(0)->second(0)->toDateString();
+        $currentHourOnUserTimezone = Carbon::now($this->timezone)->format('H:00');
```

Comment:
You should save the result of Carbon::now($this->timezone) in some variable, to be sure $currentDateOnUserTimezone and $currentHourOnUserTimezone uses the same hour, because it could be different between the two lines (by 1 µs at least ;) )

Reply:
Ohhhh this is true. Good catch.